milli/src/update/settings.rs

```diff
@@ -349,6 +349,16 @@ impl<'a, 't, 'u, 'i> Settings<'a, 't, 'u, 'i> {
     fn update_searchable(&mut self) -> Result<bool> {
         match self.searchable_fields {
             Setting::Set(ref fields) => {
+                let did_change = self
+                    .index
+                    .searchable_fields(self.wtxn)?
+                    .map(|f| f.into_iter().map(String::from).collect::<Vec<_>>())
+                    .map(|old_fields| fields != &old_fields)
+                    .unwrap_or(true); // if old_fields was None before, it was changed
+                if !did_change {
+                    return Ok(false);
+                }
```

Comment:
Can you confirm that you are retrieving the previous searchable to compare them with the new ones and if nothing changes you return false, right? It is a typical situation when users send the whole set of settings and don't care of the previous ones...

Reply:
That is correct! Would you like me to break apart that chained logic to make it more readable?

Reply:
Indeed, it would probably make this a lot more readable 😅